src/main/java/com/selfxdsd/selfweb/api/AllTasks.java

```diff
@@ -0,0 +1,258 @@
+/**
+ * Copyright (c) 2020-2021, Self XDSD Contributors
+ * All rights reserved.
+ * <p>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to read the Software only. Permission is hereby NOT GRANTED to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software.
+ * <p>
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.selfxdsd.selfweb.api;
+
+import com.selfxdsd.api.*;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Lists all Contract tasks, active and invoiced, ordered by newest assignment
+ * date.
+ * <br/>
+ * Should be used only for listing via its iterator, otherwise will
+ * throw UnsupportedOperationException.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.1
+ */
+class AllTasks implements Tasks {
```

Comment:
@criske Why do we need this class, what does it do exactly? :D 

Also, can you write a puzzle for adding unit tests for it, if we need it?

Reply:
@amihaiemil First of all this class is needed to `JsonTasks`, since it accepts only `Tasks`.
Then this class makes possible to list all contract invoiced tasks (closed taks) along with the active tasks (see the iterator implementation).
But the requirement was to show a task being active or closed with invoice id too. This is where `StatusTask` decorator comes. In iterator the active tasks are mapped to a `StatusTask.Active` impl where invoiced tasks are mapped to a `StatusTask.Closed` impl.
To display a task as active or closed(with invoice) I've just overriden `toString()` of  `StatusTask` impls following this convention: `active:-` or `closed:<invoiceId>`, which is kinda hacky imo :D . And finally these are parsed by `JsonTask` if task is `StatusTask`.

And about testing, I think I'll test the whole tasks endpoint. I don't see any test targeting it.